turtle_graphics/turtle.js

```diff
@@ -0,0 +1,324 @@
+//  the turtle has walked over.
+//
+///////////////////////////////////////////////////////////////////////////////////
+
+console.log(new Turtle(0, 4).forward(3).left().forward(3).allPoints());
+const flash = new Turtle(0, 4).forward(3).left().forward(3);
+console.log(flash.allPoints());
+
+
+///////////////////////////////////////////////////////////////////////////////////
+//
+//  Print
+//  Create a print method that draws the path that the turtle walked over as a string
+//  and logs it to the console. You should use the array of coordinates returned by 
+//  .allPoints() as your starting point. 
+//
+///////////////////////////////////////////////////////////////////////////////////
+
+new Turtle(0, 4)
+.forward(3)
+.left()
+.forward(3) 
+.right()
+.forward(5)
+.right()
+.forward(8)
+.right()
+.forward(5)
+.right()
+.forward(3)
+.left()
+.forward(3)
+.print();
+
+///////////////////////////////////////////////////////////////////////////////////
+//
+//  Stretch
+//  Make the turtle graphics program usable as a script. 
+//  It should take a string as a an argument that is seperated by dashes 
+//  (i.e. ->). This string will contain all turtle commands in abbreviated form:
+//  tX,Y for new Turtle where X & Y are numbers representing the starting x & y
+//  coordinates. If this command is not given, begin the turtle at (0, 0).
+//  fN for forward where N is a number representing how many units the turtle moves forward.
+//   r for right
+//  l for left
+//
+///////////////////////////////////////////////////////////////////////////////////
+
+$ node turtle.js t5,5-f10-r-f5-r-f10-r-f5-r-f2-r-f5-l-f2-l-f5
+ 
+$ node turtle.js f10-r-r-f10-l-f5-l-f10-r-f5-r-f11
+
+$ node turtle.js t5,5-f10-r-f5-r-f10-r-f5-r-f2-r-f5-l-f2-l-f5
+
+///////////////////////////////////////////////////////////////////////////////////
+//
+//  Save To a File  Have your script accept an  --output=[filename<]/code>
+//  where [filename] corresponds to the name of a file. 
+//  If the option is used, write the turtle drawing to the file using 
+//  fs.writeFile. Notify the user of that the write was completed.
+//
+///////////////////////////////////////////////////////////////////////////////////
+
+$ node turtle.js --output=drawing.txt f10-r-f10-r-f10-r-f10
+
+*/
+
+
+if(!process.argv[2]){
+    new Turtle(0, 0).print();
+}else{    
+    let file_name='';
+    if(process.argv[2].substr(0,9)==="--output="){
+        file_name=process.argv[2].substr(9);       
+        if(process.argv[3]){
+            process.argv[2]=process.argv[3];
+        }
+    }else if(process.argv[3] && process.argv[3].substr(0,9)==="--output="){
+        file_name=process.argv[3].substr(9);
+    }
+
+    let run_string='';
+    if(process.argv[2].indexOf('t')){
+        run_string='flash = new Turtle()';        
+    } 
+    const arg_array = process.argv[2].split("-");  
+    for (let arg of arg_array) {   
+        if (arg.substr(0,1)  === 't') {             
```

Comment:
alternative to `substr(0, 1)` in this case would be `startsWith` (just a suggestion of another method you could've used 😃 )

Reply:
It is good to know  startsWith ,the code will simple like the following, thanks.

 if(arg.startsWith('t')){    
            const pos_array = arg.substr(1).split(",");             
            run_string += " flash = new Turtle("+pos_array[0]+","+pos_array[1]+")";                        
        } else if (arg.startsWith('f')) {
            run_string +=".forward("+arg.substr(1)+")";
        } else if (arg.startsWith('r')) {
            run_string +=".right()";
        } else if (arg.startsWith('l')) {
            run_string +=".left()";
        }